src/sheet/components/SheetDataEditorCell.tsx

```diff
@@ -110,8 +118,11 @@ export default function SheetDataEditorCell({
       allData
     );
 
+    const { sheetId, sheetColumnId } = columnDefinition.typeArguments;
+    const labelDict = enumLabelDict[sheetId][sheetColumnId];
+
     const selectOptions = referenceData.map((value) => ({
-      label: value,
+      label: labelDict[value] ?? value,
```

Comment:
Cannot labelDict be undefined here, since from what I understand you only add columns that are enum columns - so for others enumLabelDict[sheetId][sheetColumnId] would not exist AFAIU